File: src/ltd/rust_lang/utils/MyList.java
package ltd.rust_lang.utils;

public interface MyList<E> {

    int ELEMENT_NOT_FOUND = -1;

    boolean isEmpty();

    void insert(int index, E element);

    boolean add(E element);

    E remove(int index);

    E set(int index, E element);

    E get(int index);

    void clear();

    boolean contains(E element);

    int indexOf(E element);

    int size();

    E remove(E element);
}

File: src/main/kotlin/wiki/mdzz/utils/list/MyList.kt
package wiki.mdzz.utils.list

interface MyList<E> {
    val isEmpty: Boolean

    fun add(index: Int, element: E?): Boolean
    fun add(element: E?): Boolean
    fun remove(index: Int): E?
    operator fun set(index: Int, element: E?): E?
    operator fun get(index: Int): E?
    fun clear()
    operator fun contains(element: E?): Boolean
    fun indexOf(element: E?): Int
    fun size(): Int
    fun remove(element: E?): E?
}
